src/langcheck/metrics/de/reference_free_text_quality.py

```diff
@@ -97,44 +95,135 @@ def sentiment(
+    score. If a score could not be computed, `None` is inserted to the score
+    and explanation lists.
+
+    Ref:
+        https://platform.openai.com/docs/guides/gpt/function-calling
+
+    Args:
+        generated_outputs: A list of model generated outputs to evaluate
+        client_type: The type of OpenAI client ('openai' or 'azure_openai')
+        client: (Optional) OpenAI or AzureOpenAI client. If this is None, we
+            will attempt to create a default client depending on the
+            ``client_type``.
+        openai_args: (Optional) Dict of additional args to pass in to the
+            ``client.chat.completions.create`` function
+
+    Returns:
+        score_list: a list of scores
+        explanation_list: a list of explanations for the scores
+    '''
+
+    def _prompt(gen_output: str) -> str:
+        return f'''
+         Sie bewerten die Stimmung einer eingereichten Aussage. Hier sind die
+        Daten:
+        [BEGINN DATEN]
+        ************
+        [Benutzeranfrage]: {gen_output}
+        ************
+        [ENDE DATEN]
+
+        Bestimmen Sie die vorherrschende Stimmung der eingereichten Aussage. Die
+        verfügbaren Bewertungen sind:
+        `Positiv` - Die eingereichte Aussage hat überwiegend eine positive
+        Stimmung
+        `Negativ` - Die eingereichte Aussage hat überwiegend eine negative
+        Stimmung
+        `Neutral` - Die eingereichte Aussage hat weder eine positive noch
+        negative Stimmung
+
+        Atmen Sie tief durch und bearbeiten Sie dieses Problem Schritt für
+        Schritt.
+        '''
+
+    def _function_call_prompt(long_assessment: str) -> str:
+        return f'''
+        Folgendes ist eine Bewertung zur Stimmung einer Aussage:
+        ************
+        [Bewertung]: {long_assessment}
+        ************
+
+        Speichern Sie die resultierende Bewertung. Die verfügbaren Bewertungen
+        sind:
+        `Positiv`
+        `Neutral`
+        `Negativ`
+        '''
+
+    sentiment_assessment_to_score = {
+        'Positiv': 1.0,
+        'Neutral': 0.5,
+        'Negativ': 0.0
+    }
+    oai_evaluator = OpenAIBasedEvaluator(
+        assessment_to_score_mapping=sentiment_assessment_to_score,
+        function_name='save_sentiment_assessment',
+        function_description="Saves a statement's sentiment assessment.",
+        argument_name='sentiment',
+        argument_description='The sentiment assessment of the statement',
+        client_type=client_type,
+        client=client,
+        openai_args=openai_args)
+
+    score_list = []
+
+    explanation_list = []
+    for gen in tqdm_wrapper(generated_outputs):
+        score, explanation = oai_evaluator.get_score(_prompt(gen_output=gen),
+                                                     _function_call_prompt)
+        score_list.append(score)
+        explanation_list.append(explanation)
+    return score_list, explanation_list
+
+
 def fluency(
```

Comment:
Fluency also has an OpenAI version, so for that one, we can pass in the German text directly.

Reply:
ah yes, I missed that, adding that too...